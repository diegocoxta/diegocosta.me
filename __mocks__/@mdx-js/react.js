module.exports = {
  MDXProvider: jest.fn(({ children }) => children),
};
