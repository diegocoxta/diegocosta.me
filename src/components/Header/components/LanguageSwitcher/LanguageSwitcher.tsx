import React from 'react';
import styled from 'styled-components';
import { useLocale, Link } from '~/hooks/useLocale';

const List = styled.div`
  display: flex;
  margin: 40px 0 0 0;
  justify-content: flex-start;
`;

const LanguageLink = styled(Link)`
  margin: 0 5px;
  color: ${({ theme }) => theme.textColor};
  font-weight: 700;
  text-decoration: none;
  font-size: 22px;
`;

export default function LanguageSwitcher() {
  const locale = useLocale();

  return (
    <List>
      {locale.getAllLanguages().map((language: string) => (
        <LanguageLink
          aria-label={`${locale.getTranslationFor('Change the language to', 'header')} ${locale.getTranslationFor(
            language
          )}`}
          key={`footer-languages-${language}`}
          to={locale.getOriginalPath()}
          language={language}
        >
          {locale.getCurrentLanguage() === language && '/'}
          {language}
        </LanguageLink>
      ))}
    </List>
  );
}
