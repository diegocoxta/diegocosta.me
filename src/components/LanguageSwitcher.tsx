import React from 'react';
import styled from 'styled-components';
import { graphql } from 'gatsby';
import * as i18Next from 'gatsby-plugin-react-i18next';

type useLocaleHook = {
  getTranslationFor: (key: string) => string;
  getCurrentLanguage: () => string;
  getAllLanguages: () => string[];
  getOriginalPath: () => string;
};

const Container = styled.div`
  display: flex;
  margin: 40px 0 0 0;
  justify-content: flex-start;
`;

const LanguageLink = styled(Link)`
  margin: 0 5px;
  color: ${({ theme }) => theme.textColor};
  font-weight: 700;
  text-decoration: none;
  font-size: 22px;
`;

export const query = graphql`
  fragment LanguageInformation on LocaleConnection {
    edges {
      node {
        ns
        data
        language
      }
    }
  }
`;

export function useLocale(): useLocaleHook {
  const { t } = i18Next.useTranslation();
  const context = i18Next.useI18next();

  function getTranslationFor(key: string) {
    return t(key);
  }

  function getCurrentLanguage() {
    return context.language;
  }

  function getAllLanguages() {
    return context.languages;
  }

  function getOriginalPath() {
    return context.originalPath;
  }

  return {
    getTranslationFor,
    getCurrentLanguage,
    getAllLanguages,
    getOriginalPath,
  };
}

export function getContentLanguage(locale: useLocaleHook, langkey?: string | null) {
  if (!langkey) {
    return;
  }

  const languagePrefix = locale.getTranslationFor('article.languagePrefix');
  const languageName = locale.getTranslationFor(`languages.${langkey}`);

  return `${languagePrefix} ${languageName}`;
}

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export function Link(props: any) {
  return <i18Next.Link {...props} />;
}

export default function LanguageSwitcher() {
  const locale = useLocale();

  return (
    <Container>
      {locale.getAllLanguages().map((language: string) => (
        <LanguageLink
          aria-label={`${locale.getTranslationFor('languageswitcher.toggle')} ${locale.getTranslationFor(
            `languages.${language}`
          )}`}
          key={`footer-languages-${language}`}
          to={locale.getOriginalPath()}
          language={language}
        >
          {locale.getCurrentLanguage() === language && '/'}
          {language}
        </LanguageLink>
      ))}
    </Container>
  );
}
