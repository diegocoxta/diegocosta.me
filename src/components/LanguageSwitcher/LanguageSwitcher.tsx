import React from 'react';
import styled from 'styled-components';
import { usei18n, Link } from '~/helpers/i18n';

import Container from '~/components/Container';

const List = styled.div`
  display: flex;
  margin: 40px 0 0 0;
  justify-content: flex-end;
`;

const LanguageLink = styled(Link)`
  margin: 0 5px;
  text-transform: uppercase;
  color: ${({ theme }) => theme.textColor};
  font-weight: 700;
  text-decoration: none;
`;

export default function LanguageSwitcher() {
  const i18n = usei18n();

  const currentLanguage = i18n.getCurrentLanguage();
  const allLanguages = i18n.getAllLanguages();

  return (
    <Container>
      <List>
        {allLanguages.map((language) => (
          <LanguageLink key={`footer-languages-${language}`} to={i18n.getOriginalPath()} language={language}>
            {currentLanguage === language && '/'}
            {language}
          </LanguageLink>
        ))}
      </List>
    </Container>
  );
}
