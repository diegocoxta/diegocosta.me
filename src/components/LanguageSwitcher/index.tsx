import { graphql } from 'gatsby';

export { default } from './LanguageSwitcher';

export const query = graphql`
  fragment LanguageInformation on LocaleConnection {
    edges {
      node {
        ns
        data
        language
      }
    }
  }
`;
