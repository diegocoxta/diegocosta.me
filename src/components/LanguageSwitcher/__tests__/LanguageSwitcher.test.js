import React from 'react';
import { render } from '@testing-library/react';

import LanguageSwitcher from '../LanguageSwitcher';

describe('<LanguageSwitcher />', () => {
  it('should render properly', () => {
    const { baseElement } = render(<LanguageSwitcher />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should render all mocked languages', () => {
    const { getByText } = render(<LanguageSwitcher />);
    expect(getByText('en')).toBeTruthy();
    expect(getByText('es')).toBeTruthy();
    expect(getByText('/pt')).toBeTruthy();
  });
});
