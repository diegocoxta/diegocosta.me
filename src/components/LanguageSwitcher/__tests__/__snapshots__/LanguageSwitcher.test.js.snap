// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LanguageSwitcher /> should render properly 1`] = `
.c0 {
  max-width: 800px;
  margin: 0 auto;
  padding: 0 20px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 40px 0 0 0;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c2 {
  margin: 0 5px;
  text-transform: uppercase;
  font-weight: 700;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<body>
  <div>
    <section
      class="c0"
    >
      <main>
        <div
          class="c1"
        >
          <a
            class="c2"
            href="/about-us"
            language="pt"
          >
            /
            pt
          </a>
          <a
            class="c2"
            href="/about-us"
            language="en"
          >
            en
          </a>
          <a
            class="c2"
            href="/about-us"
            language="es"
          >
            es
          </a>
        </div>
      </main>
    </section>
  </div>
</body>
`;
